src/libsyntax/ext/quote.rs

```diff
@@ -238,7 +238,8 @@ pub mod rt {
                 if i > 0 {
                     inner.push(TokenTree::Token(self.span, token::Colon).into());
                 }
-                inner.push(TokenTree::Token(self.span, token::Ident(segment.identifier)).into());
+                inner.push(TokenTree::Token(self.span,
+                                            token::Ident(segment.identifier, false)).into());
```

Comment:
I think the best guess here would be `is_raw(ident) = is_reserved_ident(ident) && !is_path_segment_keyword(ident)`